### PR TITLE
[dynamo][numpy] Support ndarray methods

### PR DESCRIPTION
**Pull request by larryliu0820**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #97537

 This PR adds universal support for ndarray methods. After #100839 each `NumpyNdarrayVariable` should wrap a `torch.Tensor`. This PR adds a `numpy_method_wrapper` which converts the `torch.Tensor` to `torch_np.ndarray` and then call the numpy ndarray method. Then we also try to return a `torch.Tensor` (return as-is if the value is not ndarray-like)
    

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @soumith @desertfire

### CODE REVIEW

File: torch/_dynamo/utils.py [hunk 1]
```diff
@@ -1630,6 +1630,25 @@ def numpy_attr_wrapper(obj, name):
...
+        self.method = method
+        self.__name__ = "wrapped_" + self.method
+
+    def __repr__(self):
+        return f"<Wrapped method <original {self.method}>>"
+
+    def __call__(self, *args, **kwargs):
+        obj = args[0]
+        if isinstance(obj, torch.Tensor):
+            obj = torch_np.ndarray(obj)
```
Comment (lezcano) [inline torch/_dynamo/utils.py:1646]: Why just the first variable? Shouldn't you do do the same for all the tensor variables? What if there are tuples of tensors?

**Review by lezcano — COMMENTED** [review]
This needs more tests. It just tests methods that do not take extra tensors. We should test this on methods that take tensors in the args and in the kwargs as well.

FWIW, I'm just leaving this review now, but I'm currently taking PTO, so I won't be able to follow up with further reviews, so don't wait for me.

**Review by ezyang — APPROVED** [review]

File: torch/_dynamo/utils.py (hunk 1 above)
Comment (larryliu0820) [reply torch/_dynamo/utils.py:1646]: So in torch_np they will cast all tensor like arguments to torch_np.ndarray implicitly. I can add some tests.

**Pull request closed without merging**